net/src/ipc_net_worker.rs

```diff
@@ -103,14 +108,39 @@ impl IpcNetWorker {
         if config["socketType"] != "zmq" {
             bail!("unexpected socketType: {}", config["socketType"]);
         }
+        let block_connect = config["blockConnect"].as_bool().unwrap_or(true);
         if let None = config["ipcUri"].as_str() {
+            if config["spawn"].is_object() {
+                let s = config["spawn"].as_object().unwrap();
+                if s["cmd"].is_string()
+                    && s["args"].is_array()
+                    && s["workDir"].is_string()
+                    && s["env"].is_object()
+                {
+                    let mut env = HashMap::new();
+                    for (k, v) in s["env"].as_object().unwrap().iter() {
+                        env.insert(k.to_string(), v.as_str().unwrap().to_string());
+                    }
```

Comment:
You can use an iterator and `map` + `collect` here to make this more declarative and easier to understand

Reply:
done